File: patch_supabase.py
with open('src/services/supabase.ts', 'r') as f:
    content = f.read()

new_content = content.replace(
    "const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL || 'https://placeholder.supabase.co';",
    "const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL || '';"
).replace(
    "const supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY || 'placeholder-key';",
    "const supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY || '';"
)

env_validation = """
export const isSupabaseEnvValid = Boolean(
  supabaseUrl &&
  !supabaseUrl.includes('placeholder') &&
  supabaseAnonKey &&
  !supabaseAnonKey.includes('placeholder')
);
"""

new_content = new_content.replace(
    "export const supabase = createClient(supabaseUrl, supabaseAnonKey, {",
    env_validation + "\nexport const supabase = createClient(\n  supabaseUrl || 'https://placeholder.supabase.co',\n  supabaseAnonKey || 'placeholder-key',\n  {"
)

with open('src/services/supabase.ts', 'w') as f:
    f.write(new_content)

File: patch_supabase.js
const fs = require('fs');
const content = fs.readFileSync('src/services/supabase.ts', 'utf8');

const updatedContent = content
  .replace(/const u1 = \['https:\/\/oyw', 'drriwq', 'kzsvye', 'yjxzp\.su', 'pabase\.co'\]\.join\(''\);\nconst k1 = \['sb_publis', 'hable_kn', '9Us8pmI_X2eEBv', 'K4-pRQ_noR', 'VPnFz'\]\.join\(''\);\n\nconst supabaseUrl = process\.env\.NEXT_PUBLIC_SUPABASE_URL \|\| u1;\nconst supabaseAnonKey = process\.env\.NEXT_PUBLIC_SUPABASE_PUBLISHABLE_KEY \|\| k1;/g,
  `const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL || '';\nconst supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY || '';`)
  .replace(/export const isSupabaseEnvValid = true;/g, `export const isSupabaseEnvValid = Boolean(supabaseUrl && supabaseAnonKey && supabaseUrl !== '' && supabaseAnonKey !== '');`);

fs.writeFileSync('src/services/supabase.ts', updatedContent);
